subgit/core.py

```diff
@@ -486,6 +489,108 @@ def pull(self, names):
                 log.info(f"Current commit summary on HEAD in git repo '{name}': ")
                 log.info(f"  {str(repo.head.commit.summary)}")
 
+    def delete(self, repo_names=None):
+        """
+        Method takes zero or one repo name as argument.
+
+        Checks if the repo(s) are valid git repo(s) and if there are any untracked changes.
+        """
+
+        config = self._get_config_file()
+
+        active_repos = self._get_active_repos(config)
+
+        active_repos_dict = config.get("repos", [])
+
+        if repo_names is None:  # If no names are specified in the command
+            repo_choices = ", ".join(active_repos_dict)
+
+        if isinstance(repo_names, list):  # If at least one name was specified
+            repo_choices = ", ".join(repo_names)
+
+        answer = self.yes_no(f"Are you sure you want to delete the following repos '{repo_choices}'")
+
+        if answer:
+            self._delete_repo(active_repos, repo_names)
+
+    @pysnooper.snoop()
+    def _delete_repo(self, active_repos, repos=None):
+        """
+        Helper method that recieves a list of repos to delete and takes care of the logic
+        """
+        if not repos:
+            repos = active_repos
+
+        in_conf_file = True
+
+        has_dirty_repos = False
+
+        repo_paths = []
+
+        bad_path = []
+
+        good_repos = []
```

Comment:
Unless you can comments for any of these variables, you can have them right after eachother, spacing in this case is not needed 
```suggestion
        in_conf_file = True
        has_dirty_repos = False
        repo_paths = []
        bad_path = []
        good_repos = []
```

you could also sort them by name for more OCD appleasing but also easier to read things that is in order :P